test/e2e/tests/003-surveyInfoEdit.js

```diff
@@ -0,0 +1,66 @@
+import {
+  button,
+  clearTextBox,
+  click,
+  expectExists,
+  writeIntoTextBox,
+  hoverTextBox,
+  toRightOf,
+  waitFor1sec,
+} from '../utils/api'
+import { clickHomeBtnEditSurveyInfo, waitForLoader, clickSidebarBtnHome } from '../utils/ui'
+
+const verifyHomeDashboard = async ({ label }) => {
+  await expectExists({ selector: '.home-dashboard' })
+  await expectExists({ text: label })
+}
+
+const selectors = {
+  name: { id: 'survey-info-name' },
+}
+
+describe('Survey info edit', () => {
+  test('Survey require name', async () => {
+    await waitFor1sec()
+    await clickHomeBtnEditSurveyInfo()
+
+    await clearTextBox({ selector: selectors.name })
+
+    await click(button('Save'))
+    await waitForLoader()
+
+    await hoverTextBox({ selector: selectors.name })
+    await expectExists({ text: 'Name is required' })
+
+    await waitFor1sec()
+    await clickSidebarBtnHome()
+    await verifyHomeDashboard({ label: 'Survey 1' })
+  })
+
+  test('Survey info save', async () => {
+    await clickHomeBtnEditSurveyInfo()
+
+    await clearTextBox({ selector: selectors.name })
+    await writeIntoTextBox({ text: 'survey', selector: selectors.name })
+
+    await clearTextBox({ selector: toRightOf('Label') })
```

Comment:
maybe even `toRightOf('Label')` and others can be defined among the `selectors; it can be called labelField or fieldLabel, or whatever

Reply:
yes, it's used several times